wait/sql.go

```diff
@@ -0,0 +1,66 @@
+package wait
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"github.com/docker/go-connections/nat"
+	"time"
+)
+
+//ForSQL constructs a new waitForSql strategy for the given driver
+func ForSQL(port nat.Port, driver string, url func(nat.Port) string) *waitForSql {
```

Comment:
What do you think about renaming this to `ForDB` and `waitForDB`? At the end of the day, we are waiting for the database. Using SQL for it seems an implementation detail.

Reply:
that's a good point but I assumed it'd be more consistent to point out "sql" because it can only handle sql based databases, if we just call it `forDB` the client of this function might not immediately understand it'd not work for say mongodb.

Reply:
I agree in the sense using SQL distinguish from any other storage type. My only concern was related to my mind directly connecting `waitForSQL` to `wait for the execution of an SQL sentence`... which on the other hand is true as we use `SELECT 1`.

So I'm good with this one :)